lib/ansible/modules/pip.py

```diff
@@ -611,6 +611,32 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _normalize_vcs_packages(module, package_list, pip):
```

Comment:
When adding new functions, add typing. The type checker is your friend and will help you catch typos and common mistakes early.
```suggestion
def _normalize_vcs_packages(
    module: AnsibleModule,
    package_list: list[Package],
    pip: list[str],
) -> list[Package]:
```

*(while writing this code suggestion, I had to inspect where the passed variables originate and what types of values they could hold; this helped me validate my assumtions but also, it would be very helpful if this context was already here; try using `pyrefly autotype` next time to help you get started)*

Reply:
Good stuff, I'll add type annotations to new functions moving forward. Fixed in 867450247a0fe0fbc4d293347a6d91785994c486